ospd_openvas/preferencehandler.py

```diff
@@ -659,12 +667,49 @@ def build_credentials_as_prefs(self, credentials: Dict) -> List[str]:
                 )
             # Check servic smb
             elif service == 'smb':
+                if krb5_set:
+                    self.errors.append(
+                        "Kerberos and SMB credentials are mutually exclusive."
+                    )
+                    continue
+                smb_set = True
                 cred_prefs_list.append(
                     f'{OID_SMB_AUTH}:1:entry:SMB login:|||{username}'
                 )
                 cred_prefs_list.append(
                     f'{OID_SMB_AUTH}:2:password:SMB password:|||{password}'
                 )
+            elif service == 'krb5':
+                if smb_set:
+                    self.errors.append(
+                        "Kerberos and SMB credentials are mutually exclusive."
+                    )
+                    continue
```

Comment:
Disabling both is the right movement here? I think it would be better to take one of both with higher priority (krb5 ?) instead.

Reply:
They are not both disabled. Whatever first is defined gets through. You can replicate that with the test outlined in the PR:

```
Illegal Test


check that log contains either smb or krb5 credentials, but not both
<get_scans_response status="200" status_text="OK"><scan id="97079ee9-8917-49da-aa4f-4ef95f757ac3" target="localhost" progress="100" status="finished" start_time="1733818742" end_time="1733818745"><results><result name="" type="Error Message" severity="" host="" hostname="" test_id="" port="" qod="" uri="">Malformed credential. Kerberos and SMB credentials are mutually exclusive.</result><result name="HOST_START" type="Log Message" severity="0.0" host="127.0.0.1" hostname="" test_id="" port="" qod="" uri="">Tue Dec 10 08:19:03 2024</result><result name="HOST_START" type="Log Message" severity="0.0" host="::1" hostname="" test_id="" port="" qod="" uri="">Tue Dec 10 08:19:03 2024</result><result name="SMB Test" type="Log Message" severity="0.0" host="127.0.0.1" hostname="localhost" test_id="0.0.0.0.0.0.0.0.0.42" port="general/tcp" qod="97" uri="">krb5 login:  password:  realm:  kdc:  host: localhost
</result><result name="SMB Test" type="Log Message" severity="0.0" host="127.0.0.1" hostname="localhost" test_id="0.0.0.0.0.0.0.0.0.42" port="general/tcp" qod="97" uri="">smb login: scanuser password: mypass
</result><result name="SMB Test" type="Log Message" severity="0.0" host="::1" hostname="localhost" test_id="0.0.0.0.0.0.0.0.0.42" port="general/tcp" qod="97" uri="">krb5 login:  password:  realm:  kdc:  host: localhost
</result><result name="SMB Test" type="Log Message" severity="0.0" host="::1" hostname="localhost" test_id="0.0.0.0.0.0.0.0.0.42" port="general/tcp" qod="97" uri="">smb login: scanuser password: mypass
</result><result name="HOST_END" type="Log Message" severity="0.0" host="::1" hostname="" test_id="" port="" qod="" uri="">Tue Dec 10 08:19:03 2024</result><result name="HOST_END" type="Log Message" severity="0.0" host="127.0.0.1" hostname="" test_id="" port="" qod="" uri="">Tue Dec 10 08:19:03 2024</result></results></scan></get_scans_response>⏎
```

The smb credentials went through.

Reply:
That said I changed the error message to be clearer.